ddraw/IDirect3DDeviceX.cpp

```diff
@@ -2453,11 +2542,98 @@ HRESULT m_IDirect3DDeviceX::DrawIndexedPrimitive(D3DPRIMITIVETYPE dptPrimitiveTy
 		}
 		else
 		{
+			const UINT stride = GetVertexStride(dwVertexTypeDesc);
+
+			// Handle PositionT
+			if((dwVertexTypeDesc & D3DFVF_XYZRHW) != 0 && Config.DdrawConvertHomogeneousW)
+			{
+				if(!Config.DdrawConvertHomogeneousToWorld)
+				{
+					UINT8 *vertex = (UINT8*)lpVertices;
+
+					for (UINT x = 0; x < dwVertexCount; x++)
+					{
+						float *pos = (float*) vertex;
+
+						pos[3] = 1.0f;
+
+						vertex += stride;
+					}
+
+					// Update the FVF
+					dwVertexTypeDesc = (dwVertexTypeDesc & ~D3DFVF_XYZRHW) | D3DFVF_XYZW;
+				}
+				else
+				{
+					const UINT newstride = stride - sizeof(float);
```

Comment:
Perhaps rename `newstride` into `targetStride`?

Reply:
Fixed in the latest check in.